addons/stock_picking_batch/models/stock_picking.py

```diff
@@ -323,3 +323,15 @@ def action_view_batch(self):
             'res_id': self.batch_id.id,
             'view_mode': 'form'
         }
+
+    @api.depends('partner_id')
+    @api.depends_context('display_name_partner')
+    def _compute_display_name(self):
+        res = super()._compute_display_name()
+        if not self.env.context.get('display_name_partner'):
+            return res
+        for picking in self:
+            if not picking.partner_id:
+                continue
+            picking.display_name = f"{picking.name} - {picking.partner_id.display_name}"
```

Comment:
```suggestion
            picking.display_name = f"{picking.display_name} - {picking.partner_id.display_name}"
```
to avoid erase other override